sdk/src/azure/platform/az_posix.c

```diff
@@ -26,3 +30,129 @@ AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
   (void)usleep((useconds_t)milliseconds * _az_TIME_MICROSECONDS_PER_MILLISECOND);
   return AZ_OK;
 }
+
+void az_platform_critical_error()
```

Comment:
This must be implementable by the application. We need a way to modify the behavior or allow the application to have a hook.

Either we use a `weak` symbol (so it can be redefined) or we create APIs to allow the app to take over (i.e. reboot if necessary).

Reply:
Implemented as a weak symbol, removed the implementation from az_posix.c.